lib/src/backends/record_replay/proxy.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// An object that uses [noSuchMethod] to dynamically handle invocations
+/// (property getters, property setters, and method invocations).
+abstract class ProxyObject {}
```

Comment:
While it gives you something other than `Object` or `dynamic` to throw in formal param lists, would classes annotated with `@proxy` from `dart:core` suffice?